coreclient/src/groups/process.rs

```diff
@@ -106,18 +107,26 @@ impl Group {
                     }
                 };
 
+                let sender =
+                    VerifiableClientCredential::try_from(processed_message.credential().clone())?;
+
                 // StagedCommitMessage Phase 1: Process the proposals.
 
                 // Before we process the AAD payload, we first process the
                 // proposals by value. Currently only removes are allowed.
                 for remove_proposal in staged_commit.remove_proposals() {
                     let removed_index = remove_proposal.remove_proposal().removed();
 
+                    let Some(removed_id) = self.client_by_index(connection, removed_index).await
+                    else {
+                        bail!("Unknown removed_id")
+                    };
```

Comment:
Just a style question, but you could use `context("Unknown removed_id")?;` instead of this `let else` construction.